src/mscorlib/src/System/Globalization/DateTimeFormat.cs

```diff
@@ -1003,31 +1003,35 @@ internal static string RoundTripFormat(ref DateTime dateTime)
                             offset = offset.Negate();
                         }
 
-                        Append(result, offset.Hours, 2);
+                        AppendNumber(result, offset.Hours, 2);
                         result.Append(':');
-                        Append(result, offset.Minutes, 2);
+                        AppendNumber(result, offset.Minutes, 2);
                     }
                     break;
 
                 case DateTimeKind.Utc:
                     result.Append('Z');
                     break;
-
-                default:
-                    break;
             }
 
             return StringBuilderCache.GetStringAndRelease(result);
         }
 
-        internal static void Append(StringBuilder builder, long val, int digit)
+        internal static void AppendNumber(StringBuilder builder, long val, int digits)
         {
-            if (digit > 1)
+            for (int i = 0; i < digits; i++)
             {
-                Append(builder, val / 10, digit - 1);
+                builder.Append('0');
             }
 
-            builder.Append((char)('0' + (val % 10)));
+            int index = 1;
+
+            while (val > 0 && index <= digits)
+            {
+                builder[builder.Length - index] = (char)('0' + (val % 10));
+                val = val / 10;
+                index++;
+            }
         }
```

Comment:
Given that ```AppendNumber``` is used multiple times in the same method, I would look if there is a performance improvement by aggresively inlining ```AppendNumber```. JIT may recognize it and build a single function pushing jump offsets to the stack. That would ensure that the code can avoid the call overhead.